### PR TITLE
Add PHP 8 Support

### PR DESCRIPTION
**Pull request by szainmehdi**
This PR allows installing this SDK on PHP 8.0. 

CI workflow for tests was updated to make sure tests run on PHP 8.0 as well.

Closes #116

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -33,17 +33,19 @@
         "http-interop/http-factory-guzzle": "Factory for guzzlehttp/guzzle"
     },
     "require-dev": {
-        "phpunit/phpunit": "^8.5",
-        "friendsofphp/php-cs-fixer": "^2.16",
+        "phpunit/phpunit": "^8.5 || ^9.0",
+        "bamarni/composer-bin-plugin": "^1.4.1",
```
Comment (curquiza) [inline composer.json:37]: Is this package replacement mandatory to make this meilisearch-php working with PHP 8? If it's not, can you make these changes in another PR, please?

**Review by curquiza — CHANGES_REQUESTED** [review]
Thanks for this work!

Because you change the CI jobs, could you update the `bors.toml` file at the root of this repo, and change `status = ["PHP 7.2 Test", "PHP 7.3 Test", "PHP 7.4 Test"]` into `status = ["Lint Codebase", "PHP 7.2 Test", "PHP 7.3 Test", "PHP 7.4 Test", "PHP 8.0 Test"]`?

File: composer.json (hunk 1 above)
Comment (curquiza) [reply composer.json:37]: Or even better, an issue so that we can discuss about the pros and cons of this integration before any work was done.

Comment (szainmehdi) [reply composer.json]: I've reverted these changes.

I felt that moving php-cs-fixer out of this package's dev-dependenies using bamarni/composer-bin-plugin was the best move as discussed here (https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702). However, I understand the concern and desire to have a discussion about this approach.

**Comment by szainmehdi** [comment]
I've made the requested changes! Thanks for reviewing in such a timely manner! ❤️

File: composer.json [hunk 2]
```diff
@@ -33,17 +33,17 @@
...
-        "phpunit/phpunit": "^8.5",
+        "phpunit/phpunit": "^8.5 || ^9.0",
         "friendsofphp/php-cs-fixer": "^2.16",
         "guzzlehttp/guzzle": "^7.1",
         "http-interop/http-factory-guzzle": "^1.0"
     },
     "scripts": {
         "lint": [
-            "./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix --verbose --config=./.php_cs.dist --diff --show-progress=estimating --dry-run --using-cache=yes --allow-risky=yes"
+            "./vendor/bin/php-cs-fixer fix --verbose --config=./.php_cs.dist --diff --show-progress=estimating --dry-run --using-cache=yes --allow-risky=yes"
```
Comment (jdreesen) [inline composer.json:43]:
```suggestion
            "./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix --verbose --config=./.php_cs.dist --diff --show-progress=estimating --dry-run --using-cache=yes --allow-risky=yes"
```

File: composer.json [hunk 3]
```diff
@@ -33,17 +33,17 @@
...
         ],
         "lint:fix": [
-            "./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix --verbose --config=./.php_cs.dist --diff --show-progress=estimating --using-cache=no --allow-risky=yes"
+            "./vendor/bin/php-cs-fixer fix --verbose --config=./.php_cs.dist --diff --show-progress=estimating --using-cache=no --allow-risky=yes"
```
Comment (jdreesen) [inline composer.json:46]:
```suggestion
            "./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix --verbose --config=./.php_cs.dist --diff --show-progress=estimating --using-cache=no --allow-risky=yes"
```

**Review by jdreesen — COMMENTED** [review]
I think this should be reverted, too.

File: composer.json (hunk 3 above)
Comment (szainmehdi) [reply composer.json]: Reverted, but this worked as well.

File: composer.json (hunk 1 above)
Comment (curquiza) [reply composer.json]: Your change suggestion could be useful, and that's why I'm really interested in reading your arguments and discussing with the other external contributors maintaining this package 🙂 Opening a new issue is a perfect place for that!
Thanks for your involvement, it's really nice to have contributors like you trying to improve the package, not only fixing it!

Comment (szainmehdi) [reply composer.json]: Love to hear that! I'll definitely open an issue. 👍

**Review by curquiza — APPROVED** [review]
LGTM!
bors merge
